sdk/storage/storage-blob/src/BlobQueryResponse.ts

```diff
@@ -0,0 +1,506 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { HttpResponse, isNode } from "@azure/core-http";
+
+import {
+  BlobDownloadResponseModel,
+  BlobType,
+  CopyStatusType,
+  LeaseDurationType,
+  LeaseStateType,
+  LeaseStatusType,
+  BlobDownloadHeaders,
+  BlobQueryResponseModel
+} from "./generatedModels";
+import { Metadata } from "./models";
+import { BlobQuickQueryStream, BlobQuickQueryStreamOptions } from "./utils/BlobQuickQueryStream";
+
+/**
+ * ONLY AVAILABLE IN NODE.JS RUNTIME.
+ *
+ * BlobQueryResponse implements BlobDownloadResponseModel interface, and in Node.js runtime it will
+ * parse avor data returned by blob query.
+ *
+ * @export
+ * @class BlobQueryResponse
+ * @implements {BlobDownloadResponseModel}
+ */
+export class BlobQueryResponse implements BlobDownloadResponseModel {
```

Comment:
>BlobDownloadResponseModel [](start = 42, length = 25)

Why don't we implements the underlying BlobQueryResponse?
Note that with the rehydrate priority feature,  underlying `BlobQueryResponse` and `BlobDownloadResponseModel` now diverge.